### ISSUE TITLE
"Check if generated files are up to date" job misses `regen-unicodedata` and `regen-limited-abi`

### ISSUE THREAD

**Issue by sobolevn** [python/cpython#126524]
# Feature or enhancement

Currently our `regen-all` is missing several `regen-` targets:

```
@echo "Note: make regen-stdlib-module-names, make regen-limited-abi, "
@echo "make regen-configure, make regen-sbom, and make regen-unicodedata should be run manually"
```

While our CI job to check regenerated files does not check them: https://github.com/python/cpython/blob/2a6b6b33dfe0f3c435abf2829b62ef3f1ef12cd3/.github/workflows/build.yml#L120-L124

This does not seem right, I think that we should add:
- `make regen-unicodedata` (it requires internet access, but GitHub jobs always have access, see https://github.com/python/cpython/pull/112120)
- `make regen-limited-abi` (needs to compile python to get the ABI)

How long does it work? Testing locally first:

```
make regen-unicodedata  9.44s user 0.47s system 27% cpu 35.719 total
make regen-limited-abi  2.35s user 0.76s system 77% cpu 4.019 total (with pre-existing cache)
```

I propose adding these two `make` targets to the test to be extra safe.
What do others think? :)
If agreed, I will send a PR.

### Linked PRs
* gh-126682
* gh-127594
* gh-127595
* gh-138197
* gh-138218

**Comment by sobolevn** [comment]
I don't touch `regen-configure` here, because it will require a different solution, will create a new issue about it later.

**Comment by encukou** [comment]
There's no need to add `regen-limited-abi`, since the CI runs `make check-limited-abi`.
That checks that `regen-limited-abi` wouldn't make changes, and tries to generate a nicer message than you'd get just by comparing the resulting diff.

**Comment by sobolevn** [comment]
When I add `make regen-unicodedata` here's how the CI speed changes:
- Before: `1m 59s` https://github.com/python/cpython/actions/runs/11717879421/job/32638269138?pr=126526
- After: `2m 11s` https://github.com/python/cpython/actions/runs/11777998050/job/32803523310?pr=126682

Does not seem like it is slow now.

**Comment by encukou** [comment]
Turns out `regen-unicodedata` can fail on a network issue: https://github.com/python/cpython/actions/runs/17265732493/job/48997259329?pr=138142
And it's been doing so consistently. Perhaps Unicode has scraper protection now. Downloading the data on every CI run isn't a very nice thing for us to do, anyway.

**Comment by sobolevn** [comment]
I will revert this change, sorry that it did not turn out as a net positive :(

**Comment by hugovk** [comment]
> And it's been doing so consistently.

No objection to reverting, but of the recent runs, it only failed once due to a network issue. The others were genuine test failures.

**Comment by hugovk** [comment]
I think we're done here. Thanks for the attempt!

**Comment by encukou** [comment]
> of the recent runs, it only failed once due to a network issue. 

Indeed, the failure was *not* consistent. Sorry for the misinformation.

Even so, we don't need to run this step every time. AFAIK, the only thing it checked was that the results of the unicodedata regen scripts are bit-by-bit reproducible.
